File: app/src/main/java/workshop1024/com/xproject/utils/ViewUtils.java
package workshop1024.com.xproject.utils;

import android.view.View;

public class ViewUtils {

    public static int getRelativeLeft(View view) {
        if (view.getParent() == view.getRootView()) {
            return view.getLeft();
        } else {
            return view.getLeft() + getRelativeLeft((View) view.getParent());
        }
    }

    public static int getRelativeTop(View view) {
        if (view.getParent() == view.getRootView()) {
            return view.getTop();
        } else {
            return view.getTop() + getRelativeTop((View) view.getParent());
        }
    }

}

File: android/app/src/main/java/workshop1024/com/xproject/utils/ViewUtils.kt
package workshop1024.com.xproject.utils

import android.view.View

object ViewUtils {
    fun getRelativeLeft(view: View): Int {
        return if (view.parent === view.rootView) {
            view.left
        } else {
            view.left + getRelativeLeft(view.parent as View)
        }
    }

    fun getRelativeTop(view: View): Int {
        return if (view.parent === view.rootView) {
            view.top
        } else {
            view.top + getRelativeTop(view.parent as View)
        }
    }
}